### ISSUE TITLE
Not properly error message if we don't have grants for space

### ISSUE THREAD

**Issue by ArtDu** [tarantool/tarantool#9212]
**Bug description**

If we didn't have grants to space and tried to get something from space(e.g. space:format), we'd have this error, that doesn't say us anything about grants:
```
Space '#512' does not exist
```

* OS: MacOS
* Architecture: amd64

```
Tarantool Enterprise 2.11.0-0-gc9673ebb7-r563-gc64
```

**Steps to reproduce**

Run two tarantool instances iteractively.

1. Type this in interactive console 1:
```
box.cfg{listen = 3301}
box.schema.user.create('test', { password = 'test', if_not_exists = true })
box.schema.user.grant("test","execute", 'universe')
mref = box.schema.create_space('mref')
mref:create_index('pri')
function f() return box.space.mref:format() end
f()
```

1. Type this in interactive console 2:
```
conn = require('net.box').connect('test:test@localhost:3301')
conn:call('f') -- and you'll get "error: Space '#512' does not exist"
```

> A clear and concise description of what you expected to happen.

I want to have more properly error message, because it could be not obviously if you get this in more complicated system with many instances, clients and other systems

**Comment by locker** [comment]
Blocked by new error subsystem refactoring: https://github.com/tarantool/roadmap-internal/issues/256

**Comment by Totktonada** [comment]
However, it prevents space names enumeration by an unauthorized user.

**Comment by locker** [comment]
> However, it prevents space names enumeration by an unauthorized user.

We could change the error to `Access to space 'mref' is denied`. Moreover, it'd be more correct security-wise. However, I think what the reporter wants is an error stack: `Failed to call function 'f'` *caused by* `Space 'mref' not found`. Error stacking is already supported, but it isn't serialized properly. This will be fixed in #9105.

**Comment by ArtDu** [comment]
> However, I think what the reporter wants is an error stack: Failed to call function 'f' caused by Space 'mref' not found. Error stacking is already supported, but it isn't serialized properly

Yeap, we got only that space doesn't exist in the log.
And we were frustrated cause we checked and know that space exists.

**Comment by nshy** [comment]
We have `'access denied'` error messages before the commit 014559466f8caf91512ad8920a20dac544683b28 but then decided to have `'not found'` error messages.

```
    security: Use system views instead of system spaces

    System views are used instead of direct reads of corresponding system
    spaces to explore all accessible objects such as spaces, functions, users
    and e.g. An operation with an inaccessible object produces a 'not found'
    error even if the object exists.

    In scope of #3250
```

**Comment by locker** [comment]
After a discussion, we've agreed to drop all box privilege checks from Lua code: https://github.com/tarantool/tarantool/issues/11690.

**Comment by sergepetrenko** [comment]
Superseded by #11690
